tests/test_plain.py

```diff
@@ -0,0 +1,123 @@
+# Copyright Mozilla Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from importlib.resources import files
+from textwrap import dedent
+from unittest import TestCase
+
+from moz.l10n.plain import plain_parse, plain_serialize
+from moz.l10n.resource import Entry, Resource, Section
+
+# Show full diff in self.assertEqual. https://stackoverflow.com/a/61345284
+# __import__("sys").modules["unittest.util"]._MAX_LENGTH = 999999999
+
+source = files("tests.data").joinpath("messages.json").read_bytes()
+
+
+class TestPlain(TestCase):
+    def test_parse(self):
+        res = plain_parse(source)
+        self.assertEqual(
+            res,
+            Resource(
+                [
+                    Section(
+                        [],
+                        [
+                            Entry(["SourceString", "message"], "Translated String"),
+                            Entry(["SourceString", "description"], "Sample comment"),
+                            Entry(
+                                ["MultipleComments", "message"],
+                                "Translated Multiple Comments",
+                            ),
+                            Entry(
+                                ["MultipleComments", "description"], "Second comment"
+                            ),
+                            Entry(
+                                ["NoCommentsorSources", "message"],
+                                "Translated No Comments or Sources",
+                            ),
+                            Entry(
+                                ["placeholders", "message"],
+                                "Hello$$$ $1YOUR_NAME$ at $2",
+                            ),
+                            Entry(["placeholders", "description"], "Peer greeting"),
+                            Entry(
+                                [
+                                    "placeholders",
+                                    "placeholders",
+                                    "1your_name",
+                                    "content",
+                                ],
```

Comment:
Interesting, I didn't even realize Pontoon already supports multiple nesting levels in key-value JSON files. Just checked and... it does.